src/AppBundle/Controller/CommentController.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+namespace AppBundle\Controller;
+
+use AppBundle\Entity\Article;
+use AppBundle\Entity\Author;
+use AppBundle\Entity\Comment;
+use AppBundle\Form\CommentType;
+use AppBundle\Security\CommentVoter;
+use Symfony\Component\Form\Form;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Class CommentController.
+ */
+class CommentController extends BaseController
+{
+    /**
+     * @param Request $request
+     * @param Article $article
+     * @Route("/article/comment/new/{article}", name="new_comment")
+     *
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse|Response
+     */
+    public function newAction(Request $request, Article $article)
+    {
+        /** @var Form $form */
+        $form = $this->createForm(CommentType::class);
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $comment = $form->getData();
+
+            $this->denyAccessUnlessGranted(CommentVoter::CREATE_COMMENT, $comment);
+
+            $comment->setAuthor($this->getUser()->getAuthor());
+
+            $comment->setArticle($article);
+
+            $article->addComment($comment);
+
+            $this->em()->persist($article);
+            $this->em()->persist($comment);
+
+            $this->em()->flush();
+
+            $this->get('app.notifier')
+                ->newCommentNotify($comment, $comment->getAuthor(), $article->getAuthor());
```

Comment:
You could use doctrina events in this servise and you shouldn't add this code in controllers